aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -67,7 +76,10 @@ Tensor arange(
     Scalar end,
     Scalar step,
     const TensorOptions& options) {
-  Tensor result = at::empty({0}, options);  // to be filled by arange_out
+  bool set_to_integral_dtype = !options.has_dtype() && allIntegral({start, end, step});
+  TensorOptions updated_options =
```

Comment:
This always does a copy of updated_options. We don't need to do a copy if we're not setting to integral dtype. We can avoid the copy by doing something like:
```
Tensor result;
if (set_to_integral_dtype) {
    result = at::empty({0}, options.dtype(at::ScalarType::Long));
} else {
    result = at::empty({0}, options);
}
return at::arange_out(result, start, end, step);
```